polar/src/vm.rs

```diff
@@ -1371,8 +1376,25 @@ impl PolarVirtualMachine {
         args: Vec<Term>,
     ) -> PolarResult<QueryEvent> {
         assert_eq!(args.len(), 2);
-        let left_term = self.deref(&args[0]);
-        let right_term = self.deref(&args[1]);
+        let mut left_term = self.deref(&args[0]);
+        let mut right_term = self.deref(&args[1]);
+
+        // Coerce booleans to integers.
+        fn to_int(x: bool) -> i64 {
+            if x {
+                1
+            } else {
+                0
+            }
+        }
+        if let Value::Boolean(x) = left_term.value() {
```

Comment:
Why did you add this?

Reply:
In Python, `bool` is a subclass of `int`, and so rules specialized on `Integer` are also applicable to instances of `Boolean` (i.e., `true` and `false`). Coercing booleans to ints here (and we should consider it for arithmetic operations, too) lets everything work in a natural way instead of throwing horrible errors.

Reply:
Does this behavior make sense in other languages besides Python?

Reply:
I like languages that do this particular coercion. Encoding `true` & `false` as `1` and `0` is very common across many domains, so it comes in handy. I can't think of a case off-hand where it would fail to make sense, but I guess it depends on how strict you like your typing.